zerver/webhooks/gitlab/view.py

```diff
@@ -132,15 +132,35 @@ def get_merge_request_event_body(payload: Dict[str, Any], action: str,
 
 def get_merge_request_open_or_updated_body(payload: Dict[str, Any], action: str,
                                            include_title: bool=False) -> str:
+    '''
+    For MR updated events, look at what was updated/changed and set include_description
+    to True if and only if the thing changed was the body of the MR. 
+    For MR opened events, include_description will always be True.
+     '''
+    if 'changes' in payload:
+        changes_section = payload['changes']                    
+    if action == "created":                                         
+        include_description = True                      
+    elif changes_section.get("description"):        
+        include_description = True
+    else:
+        include_description = False
+    description = None
+    if ('changes' in payload) and ('description' in payload['changes']):
+        description = payload['changes']['description']['current']
+    else:
+        description = payload['object_attributes']['description']
+        
     pull_request = payload['object_attributes']
+
     return get_pull_request_event_message(
         get_issue_user_name(payload),
         action,
         pull_request.get('url'),
         pull_request.get('iid'),
         pull_request.get('source_branch'),
         pull_request.get('target_branch'),
-        pull_request.get('description'),
+        message=description if include_description else None,
```

Comment:
To be completely honest with you, all in all this code is _really_ messy. I would suggest using something like the following instead:
```python3
description = None
changes = payload.get("changes", {})
if action == "created":
    description = payload["object_attributes"]["description"]
elif "description" in changes:
    description = payload["changes"]["description"]["current"]


...


- pull_request.get('description'),
+ message=description,
```

So you may be wondering "what exactly makes the above code cleaner?", to which I say:
1. Not introducing unnecessary boolean flags when we can just use the default value of the argument instead.
2. Not performing the same check multiple times needlessly.
3. Smaller footprint and increased elegance. All we want to do here is potentially change the description... that's it. We shouldn't need to complicate the code by much. It should be as simple as "Were there changes? Was the description changed? Ok."

Note: if there are changes and the changed thing is thing is the description we can safely assume that the "current" key would be present. If it wasn't then we would actually want the code to raise an exception so that we could see why the key isn't present (which would be _highly_ unlikely).

Reply:
I got the test fixtures by using  https://webhook.site/ to capture the webhook payloads.



I tried to solve the issue based on the way you suggested earlier using the include_description param,

 if you want to fix this issue, you'll need to modify the `get_merge_request_open_or_updated_body` method in `zerver/webhooks/gitlab/view.py`. Simply put, the fix will be to add a new parameter to the method similar to `include_title` but for the description instead: `include_description`. For MR updated events, look at what was updated/changed and set `include_description` to `True` if and only if the thing changed was the body of the MR. For MR opened events, `include_description` will always be `True`.

I see that the new suggestions no longer requires the use of this boolean flag. I will update accordingly. Thank you 👍

Reply:
> I got the test fixtures by using https://webhook.site/ to capture the webhook payloads.

If you did that, then what's up with all of the `Tomasz Kolek` and `https://gitlab.com/tomaszkolek0/my-awesome-project` stuff?

Reply:
Also, that earlier `include_description` suggestion was just a tentative thing to give you an idea, always feel free to use your own judgment to make the best decision 👍🏽.